### ISSUE TITLE
Failure after one call to rcloud.update_notebook()

### ISSUE THREAD

**Issue by dougmet** [att/rcloud#1526]
Decided to post this issue as I'm seeing it on a couple of independent RCloud instances now. This is affecting 1.3.4 and 1.4.

If you do anything that calls rcloud.update_notebook(), such as changing the text of a cell or adding an asset, then no further calls to RServe are ever sent, further edits are not saved and you can't execute a cell. From the RServe logs it looks like the first request is responded to but I don't know how to get more verbose error messages than that.

Is there anything you could recommend on the JS side to find an error message? I appreciate I'm not giving you much to go on.

Along with an Ubuntu server that has been gradually updated through the versions I'm also having this problem using the setup in this vagrant file + provisioner https://github.com/dougmet/vagrantRcloud.

Thanks.

**Comment by gordonwoodhull** [comment]
Thanks for the report @dougmet. Let's try to narrow in on this a bit more, since obviously this is not happening in all cases.

Is this a particularly large notebook? Or is it possible there is invalid text in the notebook? Or do you mean you encounter this with all notebooks?

I take it you are not seeing any feedback as all in the session pane. To diagnose it better in JavaScript, see if the browser's debugger console shows any errors.

**Comment by dougmet** [comment]
Hi @gordonwoodhull , thanks for looking at this.

Actually yes, for one of the instances (which is on the same network as its Enterprise GitHub) I do find that the problem is much worse for larger notebooks. I often have one or two assets as well. For smaller notebooks it seems more robust. That's a good lead.

For my local version at home (which connects to public GitHub) I find this problem with all notebooks. So perhaps the slower response?

I don't get any special errors in the console. Sometimes a warning about keyboard events and the ace editor but I don't feel like that's going to be the issue.

**Comment by gordonwoodhull** [comment]
It doesn't even tell you it has disconnected? I am not sure if we are talking about the same bug.

@s-u, is it now reasonably easy to install the "proxified" server now, to see if it's more stable and/or aide in troubleshooting?

**Comment by dougmet** [comment]
No disconnection message. In the case of running an R cell you still get the spinning running indicator but it doesn't appear to send anything out (or perhaps RServe doesn't take it).

**Comment by gordonwoodhull** [comment]
Do any changes to the notebook get lost? 

I'm trying to ascertain if something is failing or if the connection is just hanging somehow.

We have a proxy server in 1.5 which may help with connection stability.

**Comment by dougmet** [comment]
Yes and no. The first change you make gets saved. After that nothing is saved. I wish I could give you better information. As I say, there are no obvious errors on either the R or JS side, so in that sense it looks like it's hanging. But I don't really know.

**Comment by gordonwoodhull** [comment]
I've confirmed with @dougmet and team that they are still having these issues. Hopefully we will be able to put some effort into stability and connectivity for 1.5

**Comment by gordonwoodhull** [comment]
Disconnection issues are greatly improved with the [proxified server](https://github.com/att/rcloud/wiki/ProxifiedSetup). Unfortunately websocket still seems to be a brittle protocol over some networks.
